runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/ComputationStateCacheTest.java

```diff
@@ -0,0 +1,249 @@
+import org.apache.beam.runners.dataflow.worker.streaming.config.ComputationConfig;
+import org.apache.beam.runners.dataflow.worker.util.BoundedQueueExecutor;
+import org.apache.beam.runners.dataflow.worker.windmill.Windmill;
+import org.apache.beam.runners.dataflow.worker.windmill.state.WindmillStateCache;
+import org.apache.beam.runners.dataflow.worker.windmill.work.budget.GetWorkBudget;
+import org.apache.beam.sdk.fn.IdGenerators;
+import org.apache.beam.vendor.grpc.v1p60p1.com.google.protobuf.ByteString;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.ImmutableMap;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.ImmutableSet;
+import org.joda.time.Instant;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ComputationStateCacheTest {
+  private final BoundedQueueExecutor workExecutor = mock(BoundedQueueExecutor.class);
+  private final WindmillStateCache.ForComputation stateCache =
+      mock(WindmillStateCache.ForComputation.class);
+  private final ComputationConfig.Fetcher configFetcher = mock(ComputationConfig.Fetcher.class);
+  private ComputationStateCache computationStateCache;
+
+  private static Work createWork(long workToken, long cacheToken) {
+
+    return Work.create(
+        Windmill.WorkItem.newBuilder()
+            .setKey(ByteString.copyFromUtf8(""))
+            .setShardingKey(1)
+            .setWorkToken(workToken)
+            .setCacheToken(cacheToken)
+            .build(),
+        Instant::now,
+        Collections.emptyList(),
+        unused -> {});
+  }
+
+  @Before
+  public void setUp() {
+    computationStateCache =
+        ComputationStateCache.create(
+            configFetcher, workExecutor, ignored -> stateCache, IdGenerators.decrementingLongs());
+  }
+
+  @Test
+  public void testGet_computationStateCached() {
+    String computationId = "computationId";
+    MapTask mapTask = new MapTask().setStageName("stageName").setSystemName("systemName");
+    Map<String, String> userTransformToStateFamilyName =
+        ImmutableMap.of("userTransformName", "stateFamilyName");
+    ComputationConfig computationConfig =
+        ComputationConfig.create(mapTask, userTransformToStateFamilyName, ImmutableMap.of());
+    when(configFetcher.fetchConfig(eq(computationId))).thenReturn(Optional.of(computationConfig));
+    computationStateCache.get(computationId);
+    Optional<ComputationState> computationState = computationStateCache.get(computationId);
+    assertTrue(computationState.isPresent());
+    assertThat(computationState.get().getComputationId()).isEqualTo(computationId);
+    assertThat(computationState.get().getMapTask()).isEqualTo(mapTask);
+    assertThat(computationState.get().getTransformUserNameToStateFamily())
+        .isEqualTo(userTransformToStateFamilyName);
+  }
+
+  @Test
+  public void testGet_computationStateNotCached() {
```

Comment:
how about combining the above and this one by just verifying on both of the get calls above?

If you feel strongly about them being separate I think this one should come first since it's doing less

Reply:
moved